src/test/java/com/google/cloud/spanner/pgadapter/StatementParserTest.java

```diff
@@ -68,67 +77,72 @@ public void testRemoveCommentsAndTrim() {
             + "  INSERT VALUES (v.column1, v.column2, v.column3)\n"
             + "WHEN MATCHED\n"
             + "  UPDATE SET FirstName = v.column2,\n"
-            + "             LastName = v.column3;";
-    result = StatementParser.removeCommentsAndTrim(sqlStatement);
+            + "             LastName = v.column3";
+    result = parser.removeCommentsAndTrim(sqlStatement);
     Assert.assertEquals(result, expectedResult);
 
     // Dollar Quoted
     sqlStatement = "$$--foo$$";
-    result = StatementParser.removeCommentsAndTrim(sqlStatement);
+    result = parser.removeCommentsAndTrim(sqlStatement);
     Assert.assertEquals(result, sqlStatement);
 
     sqlStatement = "$$\nline 1\n--line2$$";
-    result = StatementParser.removeCommentsAndTrim(sqlStatement);
+    result = parser.removeCommentsAndTrim(sqlStatement);
     Assert.assertEquals(result, sqlStatement);
 
     sqlStatement = "$bar$--foo$bar$";
-    result = StatementParser.removeCommentsAndTrim(sqlStatement);
+    result = parser.removeCommentsAndTrim(sqlStatement);
     Assert.assertEquals(result, sqlStatement);
 
     sqlStatement = "SELECT FOO$BAR FROM SOME_TABLE";
-    result = StatementParser.removeCommentsAndTrim(sqlStatement);
+    result = parser.removeCommentsAndTrim(sqlStatement);
     Assert.assertEquals(result, sqlStatement);
 
     sqlStatement = "SELECT FOO$BAR -- This is a comment\nFROM SOME_TABLE";
     expectedResult = "SELECT FOO$BAR \nFROM SOME_TABLE";
-    result = StatementParser.removeCommentsAndTrim(sqlStatement);
+    result = parser.removeCommentsAndTrim(sqlStatement);
     Assert.assertEquals(result, expectedResult);
 
     // Embedded Comments
     sqlStatement =
         "/* This is a comment /* This is an embedded comment */ This is after the embedded comment */ SELECT 1";
     expectedResult = "SELECT 1";
-    result = StatementParser.removeCommentsAndTrim(sqlStatement);
+    result = parser.removeCommentsAndTrim(sqlStatement);
     Assert.assertEquals(result, expectedResult);
 
     // No effect HashTag Comment
     sqlStatement = "# this is a comment\nselect * from foo";
-    result = StatementParser.removeCommentsAndTrim(sqlStatement);
+    result = parser.removeCommentsAndTrim(sqlStatement);
     Assert.assertEquals(result, sqlStatement);
 
     sqlStatement = "select *\nfrom foo # this is a comment\nwhere bar=1";
-    result = StatementParser.removeCommentsAndTrim(sqlStatement);
+    result = parser.removeCommentsAndTrim(sqlStatement);
     Assert.assertEquals(result, sqlStatement);
 
     // When parameters are mixed with dollar-quoted string
-    sqlStatement = "$1$$?it$?s$$$2";
```

Comment:
This was not a valid PostgreSQL SQL fragment, as there is no space between the query parameter and the start/end of the dollar-quoted string. The PGAdapter parser accepted it, but it is rejected by PostgreSQL and by the parser in the Java client library.

This also applies to the SQL fragments below.